src/org/labkey/snd/SNDUserSchema.java

```diff
@@ -48,22 +48,23 @@
 
 public class SNDUserSchema extends SimpleUserSchema
```

Comment:
SNDUserSchema could implement UserSchema.HasContextualRoles.  Then all the contained tables could just call getUserSchema().getContextualRoles().

Reply:
Perfect. Done